include/alpaka/core/CallbackThread.hpp

```diff
@@ -37,71 +39,59 @@ namespace alpaka::core
                 m_thread.join();
             }
         }
-
-        // Note: due to different std lib implementations of packaged_task, the lifetime of the passed function either
-        // ends when the packaged_task is destroyed (while the returned future is still alive) or when both are
-        // destroyed.
-        template<typename NullaryFunction>
-        auto submit(NullaryFunction&& nf) -> std::future<void>
```

Comment:
A refactoring of `CallbackThread` was required too to silent the last errors in the thread sanetizer.